net/api/service.go

```diff
@@ -112,6 +112,9 @@ func (s *Service) GetThread(ctx context.Context, req *pb.GetThreadRequest) (*pb.
 	if err != nil {
 		return nil, err
 	}
+	if info.ReadKey == nil && info.FollowKey == nil {
```

Comment:
Is this the best place to perform this check? Or should we do this in `threadInfoToProto`? In theory, one shouldn't call this at all if the thread doesn't exist locally, but there is a non-zero likelihood that this will get called for a thread that doesn't exist.

Reply:
`GetThread` should now return `ErrThreadNotFound` if it's not available. Maybe you saw this before the last big PR went in?

Reply:
Ah yes indeed!